src/transformers/models/chameleon/modeling_chameleon.py

```diff
@@ -1476,6 +1675,124 @@ def set_decoder(self, decoder):
     def get_decoder(self):
         return self.model
 
+    @torch.no_grad()
+    def generate(
+        self,
+        inputs: Optional[torch.Tensor] = None,
+        generation_config: Optional[ChameleonGenerationConfig] = None,
+        logits_processor: Optional[LogitsProcessorList] = None,
+        **kwargs,
+    ) -> Union[GenerateOutput, torch.LongTensor]:
+        generation_config, model_kwargs = self._prepare_generation_config(generation_config, **kwargs)
+
+        inputs_tensor, model_input_name, model_kwargs = self._prepare_model_inputs(
+            inputs, generation_config.bos_token_id, model_kwargs
+        )
+        input_ids = inputs_tensor if model_input_name == "input_ids" else model_kwargs.pop("input_ids")
+
+        # Prepare `max_length` depending on other stopping criteria.
+        input_ids_length = input_ids.shape[-1]
+        has_default_max_length = kwargs.get("max_length") is None and generation_config.max_length is not None
+        has_default_min_length = kwargs.get("min_length") is None and generation_config.min_length is not None
+        generation_config = self._prepare_generated_length(
+            generation_config=generation_config,
+            has_default_max_length=has_default_max_length,
+            has_default_min_length=has_default_min_length,
+            model_input_name=model_input_name,
+            inputs_tensor=inputs_tensor,
+            input_ids_length=input_ids_length,
+        )
```

Comment:
Can we remove this part, as we don;t really need all these steps? This allows us to have custom `generate()` without extra maintaining efforts, because we'll be simply extending LogitsProcessor and passing it further to handle. 

For the 'generation_mode' we can simply leave the first line

```
generation_config, model_kwargs = self._prepare_generation_config(generation_config, **kwargs)
```

Otherwise, we'll have to manually change Chameleon modeling every time `generate()` is modified

Reply:
hmm... I need to pass `generation_config.max_length` to `SuppressTokensInIndexRangeLogitsProcessor` a few lines below to prevent Chameleon from starting to generate image tokens if there isn't enough space.

IMO this is the most reliable way to get `generation_config.max_length`

Reply:
Oh, right! Yes, in that case we need it